src/pluginLifeCycle.ts

```diff
@@ -0,0 +1,29 @@
+/*
+* This program and the accompanying materials are made available under the terms of the *
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at *
+* https://www.eclipse.org/legal/epl-v20.html                                      *
+*                                                                                 *
+* SPDX-License-Identifier: EPL-2.0                                                *
+*                                                                                 *
+* Copyright Contributors to the Zowe Project.                                     *
+*                                                                                 *
+*/
+
+import { AbstractPluginLifeCycle, Logger } from "@zowe/imperative";
+
+class PluginLifeCycle extends AbstractPluginLifeCycle {
+    public postInstall(): void | Promise<void> {
```

Comment:
not a required change, but this return type and the return type for `preUninstall` can be reduced to just `void`